src/groupByKey.ts

```diff
@@ -2,6 +2,25 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type ObjectForSorting = {
+  [key: string]: string | number;
+};
+
+export function groupByKey<T>(
+  items: ObjectForSorting[],
+  key: string,
+): GroupsMap<T> {
+  const group: GroupsMap<T> = {};
+
+  items.forEach((object: ObjectForSorting) => {
```

Comment:
The 'forEach' loop can be replaced with 'reduce' for better performance and readability. The 'reduce' function is more suitable for this case because it can transform an array into an object. Additionally, it will ensure compliance with the 'array-callback-return' rule in ESLint as 'reduce' requires a return value.